runConnectivityIntegrationTests

```diff
@@ -0,0 +1,299 @@
+    TESTSRCDIR="${TESTBASEDIR}/src/test/java"
+    TESTPOM="${TESTBASEDIR}/pom.xml"
+    TESTCODE="${TESTSRCDIR}/ConnectivityTest.java"
+    mkdir -p "${TESTSRCDIR}"
+    cat >"${TESTPOM}" <<EOL
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.igniterealtime.tests</groupId>
+    <artifactId>smack-bosh-example</artifactId>
+    <version>1.0</version>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.igniterealtime.smack</groupId>
+            <artifactId>smack-java7</artifactId>
+            <version>4.3.4</version>
+        </dependency>
+        <dependency>
+            <groupId>org.igniterealtime.smack</groupId>
+            <artifactId>smack-tcp</artifactId>
+            <version>4.3.4</version>
+        </dependency>
+        <dependency>
+            <groupId>org.igniterealtime.smack</groupId>
+            <artifactId>smack-bosh</artifactId>
+            <version>4.3.4</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>
+EOL
+    cat >"${TESTCODE}" <<EOL
+import org.jivesoftware.smack.ConnectionConfiguration;
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.bosh.BOSHConfiguration;
+import org.jivesoftware.smack.bosh.XMPPBOSHConnection;
+import org.jivesoftware.smack.tcp.XMPPTCPConnection;
+import org.jivesoftware.smack.tcp.XMPPTCPConnectionConfiguration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ConnectivityTest
+{
+    /**
+     * Verifies that a few connections can be established and used to authenticate over BOSH.
+     */
+    @Test
+    public void testBoshConnections() throws Exception
+    {
+        // Setup test fixture.
+        final BOSHConfiguration config = BOSHConfiguration.builder()
+                .setFile("/http-bind/")
+                .setXmppDomain("$DOMAIN")
+                .setHost("$HOST")
+                .setPort($BOSHPORT)
+                .setUseHttps(false)
+                .setSecurityMode(ConnectionConfiguration.SecurityMode.disabled)
+                .build();
+        final int iterations = 25;
+
+        // Execute system under test.
+        int result = 0;
+        for (int i=0; i<iterations; i++) {
+            final XMPPBOSHConnection con = new XMPPBOSHConnection(config);
+            con.connect();
+            try {
+                con.login("john", "secret");
+                Thread.sleep(200);
```

Comment:
This is okay, but if we're seeing it flake in the future (e.g. if the local Openfire isn't performant on the GH CI infrastructure), we could replace this magic number with a listener pattern instead.